resourseleak/TestCloseQuietly.java

```diff
@@ -0,0 +1,25 @@
+package resourceleak;
+
+import com.amazonaws.services.s3.model.S3Object;
+
+import java.io.IOException;
+
+
+public class TestCloseQuietly {
+
+    public String testCloseQuietly(String bucketName, String key)
+            throws AmazonServiceException, SdkClientException {
+        rejectNull(bucketName, "Bucket name must be provided");
+        rejectNull(key, "Object key must be provided");
+
+        S3Object object = getObject(bucketName, key);
+        try {
+            return IOUtils.toString(object.getObjectContent());
+        } catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)